chat_shell/chat_shell/tools/builtin/knowledge_base.py

```
@@ -752,3 +763,232 @@ async def _format_rag_result(
+                    }
+                )
+                source_index += 1
+
+        # Try to run async persist in event loop
+        try:
+            loop = asyncio.get_event_loop()
+            if loop.is_running():
+                # If event loop is running, create a task
+                asyncio.create_task(
+                    self._persist_rag_results(chunks_used, source_references, query)
+                )
```

Ruff (0.14.10)

Store a reference to the return value of `asyncio.create_task`

(RUF006)